gmso/external/convert_hoomd.py

```diff
@@ -0,0 +1,881 @@
+    """Parse bond forces."""
+
+    def _parse_harmonic(container, btypes):
+        for btype in btypes:
+            # TODO: Unit conversion
+            container.params["-".join(btype.member_types)] = {
+                "k": btype.parameters["k"],
+                "r0": btype.parameters["r_eq"],
+            }
+        return container
+
+    unique_btypes = top.bond_types(filter_by=PotentialFilters.UNIQUE_NAME_CLASS)
+    groups = dict()
+    for btype in unique_btypes:
+        group = potential_types[btype]
+        if group not in groups:
+            groups[group] = [btype]
+        else:
+            groups[group].append(btype)
+
+    for group in groups:
+        expected_units_dim = potential_refs[group][
+            "expected_parameters_dimensions"
+        ]
+        groups[group] = _convert_params_units(
+            groups[group], expected_units_dim, base_units
+        )
+
+    btype_group_map = {
+        "HarmonicBondPotential": {
+            "container": hoomd.md.bond.Harmonic,
+            "parser": _parse_harmonic,
+        },
+    }
+    bond_forces = list()
+    for group in groups:
+        bond_forces.append(
+            btype_group_map[group]["parser"](
+                container=btype_group_map[group]["container"](),
+                btypes=groups[group],
+            )
+        )
+
+    return bond_forces
+
+
+def _parse_angle_forces(top, potential_types, potential_refs, base_units):
+    """Parse angle forces."""
+
+    def _parse_harmonic(container, agtypes):
+        for agtype in agtypes:
+            # TODO: Unit conversion
+            container.params["-".join(agtype.member_types)] = {
+                "k": agtype.parameters["k"],
+                "t0": agtype.parameters["theta_eq"],
+            }
+        return container
+
+    unique_agtypes = top.angle_types(
+        filter_by=PotentialFilters.UNIQUE_NAME_CLASS
+    )
+    groups = dict()
+    for agtype in unique_agtypes:
+        group = potential_types[agtype]
+        if group not in groups:
+            groups[group] = [agtype]
+        else:
+            groups[group].append(agtype)
+
+    base_units["angle"] = u.degree
+    for group in groups:
+        expected_units_dim = potential_refs[group][
+            "expected_parameters_dimensions"
+        ]
+        groups[group] = _convert_params_units(
+            groups[group], expected_units_dim, base_units
+        )
+
+    agtype_group_map = {
+        "HarmonicAnglePotential": {
+            "container": hoomd.md.angle.Harmonic,
+            "parser": _parse_harmonic,
+        },
+    }
+    angle_forces = list()
+    for group in groups:
+        angle_forces.append(
+            agtype_group_map[group]["parser"](
+                container=agtype_group_map[group]["container"](),
+                agtypes=groups[group],
+            )
+        )
+
+    return angle_forces
+
+
+def _parse_dihedral_forces(top, potential_types, potential_refs, base_units):
```

Comment:
In this funciton, do we need to make sure any values for angles are in the correct units, similar to you're doing in the `_parse_angles` function?